persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/Conversion.java

```diff
@@ -73,6 +85,63 @@
     TYPE_MAPPINGS = ImmutableMap.copyOf(types);
   }
 
+  public static ByteBuffer toPagingState(PagingPosition pos, Parameters parameters) {
+    Row row = pos.currentRow();
+    Set<Pair<String, String>> tables =
+        row.columns().stream()
+            .map(c -> Pair.create(c.keyspace(), c.table()))
+            .collect(Collectors.toSet());
+
+    if (tables.isEmpty()) {
+      throw new IllegalArgumentException("Missing table information in custom paging request.");
+    }
+
+    if (tables.size() > 1) {
+      throw new IllegalArgumentException("Too many tables are referenced: " + tables);
+    }
+
+    Pair<String, String> tableName = tables.iterator().next();
+    TableMetadata table = Schema.instance.validateTable(tableName.left, tableName.right);
+
+    MultiCBuilder builder = MultiCBuilder.create(table.partitionKeyAsClusteringComparator(), false);
```

Comment:
Why not use a `CBuilder` here instead of `MultiCBuilder`?  That's from looking in the implementation of `ClusteringComparator#make()`.  If possible maybe it's worth using `ClusteringComparator#make()` instead:

```java
    List<ByteBuffer> partitionKey = new ArrayList<>(cfm.partitionKeyColumns().size());

    for (ColumnDefinition c : cfm.partitionKeyColumns()) {
      ByteBuffer value = row.getBytesUnsafe(c.name.toCQLString());

      if (value == null) {
        throw new IllegalArgumentException(
            String.format(
                "Partition key value is not present in current row (table: %s, column: %s)",
                table, c.name.toCQLString()));
      }
      partitionKey.add(value);
    }

    Clustering clustering = cfm.getKeyValidatorAsClusteringComparator().make(partitionKey.toArray());
```

It's simpler, but creates a bit more garbage.